### DISCUSSION TITLE
Nested devices double counting sub panel.

### DISCUSSION THREAD

**Discussion by ddovidenko** [jertel/vuegraf#154] [Ideas]
Really awesome script, loving it so far! Didn't see much discussion of this, but the native app avoids double counting nested device data from the 200A CTs if it knows one device is nested under another. The setup in question is when you have one Vue device with 200A CTs installed on the main breaker, and then another Vue device has 200A CTs on the breaker feeding a sub panel. The native app can handle this as it knows the primary device has the true total usage. However, with the current influx dashboard that data gets double counted. I don't think it's super hard to fix and I'm going to poke around in the dashboard queries, just seems like a potential feature to implement.

Poked around a little and it seems pretty easy to fix. For instance for the Combined Usage Gauge I just changed the filter to be more specific for my main panel:

`|> filter(fn: (r) => r["device_name"] =~ /Main Panel/ and r["device_name"] !~ /-Balance/)`

May just be something to note in the docs more than a feature since it's so simple. I'm just looking through the whole template and changing out the instances as appropriate.

**Comment by jertel** [comment]
If you had named the nested 200A CT device "Subpanel" instead of "Panel" would that have avoided the problem?

**Reply by ddovidenko** [reply]
So one was named "Main Panel" and one was "Sub Panel" the original filter:

`|> filter(fn: (r) => r["device_name"] =~ /Panel/ and r["device_name"] !~ /-Balance/)`

Was catching and combining them both. Hence why i added Main to the filter. I guess I could have just renamed the subpanel something different without the word panel in it.

**Reply by jertel** [reply]
I believe using "Subpanel", exactly with that specific casing (lowercase `p`)  would also have worked. Definitely worth mentioning in the docs.

**Reply by ddovidenko** [reply]
That's a great point! Sorry wasn't thinking about that.

Hey another question, I've enabled detailed collection, and I see the log messages in my container logs, but the influx data output doesn't seem very granular. Initially I thought the boolean detailed variable in the dashboard wasn't working, but then saw it was lagging by an hour (as expected), so that kind of confirmed I'm not getting full granularity, and even a manual query seems to show two very similar data sets:

![image](https://github.com/jertel/vuegraf/assets/36215809/286662f8-103f-416c-a864-1a73a143c11b)

My config is just:    "detailedDataEnabled": "true",

Here are the logs I see:

2024-02-08 05:41:08.897327+00:00 | INFO  | Submitting datapoints to database; account="Dmitry Dovidenko"; points=124036

124036 points divided by 36 in a set divided by 60 minutes is seems like it would yield 57 data points per minute per CT on average.

I'm just not sure why they're not making it into influx or if I'm just not finding them, haha. Thanks again for taking the time to chat. This is a really awesome project you've built!

**Reply by jertel** [reply]
The non-detailed (60 second avg) will always appear to lag the detailed data. Here's an example of the same query as yours, but against one of my always-on pumps:

![image](https://github.com/jertel/vuegraf/assets/16639611/bd27c71c-74f3-40ed-874c-2dfb89eda44a)


Notice that I had to set a custom window period of 1s to see that granular detailed data. What window period did you use?

**Reply by ddovidenko** [reply]
Yep sorry still new at using influxdb, looks like I can see them with a 1s window set in the data explorer. Have you been able to make it work with the dashboard when setting detailed to true and zooming in or setting a time range? I thought the  v.windowPeriod in the graph config would make it more granular, but doesn't seem like it's working.

**Reply by jertel** [reply]
The dashboard auto calculates the `v.windowPeriod` based on the selected time range. If you choose a 2 minute time range from a couple hours ago, with detailed dropdown set to True, then you should be able to see the graphs show full 1s granularity.

**Reply by ddovidenko** [reply]
Gotcha yea I was just reading that, I guess I figured v.windowPeriod zooms with you zooming, but it doesn't. Thanks again!
